src/components/AddNote.jsx

```diff
@@ -32,20 +32,20 @@ function AddNote() {
     timeAdded: '',
   });
 
-  const addIcon = <Icon name="add" />;
+  const addIcon = <Icon name='add' />;
 
   /**
    * attaches a unique id to a blank note defined in its state and sends to
    * Main component's dispatch function to
    */
-  const onAdd = () => {
-    const dateAdded = new Date();
-    const addTime = dateAdded.getTime();
-    mainStateDispatchFunction({ type: 'add', state: noteTemp, timeAdded: addTime });
+  const onAdd = useCallback(() => {
+    // const dateAdded = new Date();
+    const addTime = new Date().getTime();
+    mainStateDispatchFunction({ type: 'add', newNote: noteTemp, timeAdded: addTime });
     // the code below is to generate a new id for the next onAdd() event
-    const uniqueId = Math.floor(Math.random() * (10 ** 9) + 1);
+    const uniqueId = new Date().getUTCMilliseconds();
```

Comment:
About "but if the page weren't static and kept the notes on the server, it wouldn't be unique, would it? I changed it to the first approach ( new Date().getUTCMilliseconds() ) 🙂". Great question.

I am proud.
Now the answer is the following. If we were using a database (the same as keeping the notes in the server) we could make it generate an ID automatically by incrementing primary key. So we would not need any `uniqueId` in here 👍